tfe/data_source_workspace_ids_test.go

```diff
@@ -121,6 +122,106 @@ func TestAccTFEWorkspaceIDsDataSource_wildcard(t *testing.T) {
 	})
 }
 
+func TestAccTFEWorkspaceIDsDataSource_tags(t *testing.T) {
+	rInt := rand.New(rand.NewSource(time.Now().UnixNano())).Int()
+	orgName := fmt.Sprintf("tst-terraform-%d", rInt)
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:  func() { testAccPreCheck(t) },
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccTFEWorkspaceIDsDataSourceConfig_tags(rInt),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					// organization attribute
+					resource.TestCheckResourceAttr(
+						"data.tfe_workspace_ids.good", "organization", orgName),
+
+					// full_names attribute
+					resource.TestCheckResourceAttr(
+						"data.tfe_workspace_ids.good", "full_names.%", "2"),
+					resource.TestCheckResourceAttr(
+						"data.tfe_workspace_ids.good",
+						fmt.Sprintf("full_names.workspace-foo-%d", rInt),
+						fmt.Sprintf("tst-terraform-%d/workspace-foo-%d", rInt, rInt),
+					),
+					resource.TestCheckResourceAttr(
+						"data.tfe_workspace_ids.good",
+						fmt.Sprintf("full_names.workspace-bar-%d", rInt),
+						fmt.Sprintf("tst-terraform-%d/workspace-bar-%d", rInt, rInt),
+					),
+
+					// ids attribute
+					resource.TestCheckResourceAttr(
+						"data.tfe_workspace_ids.good", "ids.%", "2"),
+					resource.TestCheckResourceAttrSet(
+						"data.tfe_workspace_ids.good", fmt.Sprintf("ids.workspace-foo-%d", rInt)),
+					resource.TestCheckResourceAttrSet(
+						"data.tfe_workspace_ids.good", fmt.Sprintf("ids.workspace-bar-%d", rInt)),
+
+					// id attribute
+					resource.TestCheckResourceAttrSet("data.tfe_workspace_ids.good", "id"),
+				),
+			},
+		},
+	})
+}
+
+func TestAccTFEWorkspaceIDsDataSource_both(t *testing.T) {
+	rInt := rand.New(rand.NewSource(time.Now().UnixNano())).Int()
+	orgName := fmt.Sprintf("tst-terraform-%d", rInt)
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckTFEWorkspaceDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccTFEWorkspaceIDsDataSourceConfig_both(rInt),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					// organization attribute
+					resource.TestCheckResourceAttr(
+						"data.tfe_workspace_ids.good", "organization", orgName),
+
+					// full_names attribute
+					resource.TestCheckResourceAttr(
+						"data.tfe_workspace_ids.good", "full_names.%", "1"),
+					resource.TestCheckResourceAttr(
+						"data.tfe_workspace_ids.good",
+						fmt.Sprintf("full_names.workspace-foo-%d", rInt),
+						fmt.Sprintf("tst-terraform-%d/workspace-foo-%d", rInt, rInt),
+					),
```

Comment:
Should we have a test for for `workspace-bar` as well? The `good` tag is on both workspaces